sdcm/rest/rest_client.py

```diff
@@ -0,0 +1,48 @@
+from typing import Dict, Literal
+
+import requests
+from requests import Response
+
+from sct import LOGGER
+
+
+class RestClient:
```

Comment:
if this class exists only to add this _base_url functionality - this could be done more generic and with less code/easier by subclassing Session object like described in https://stackoverflow.com/a/51026159/3891194

Reply:
The intention here is to have a base RestClient class, so that more specific clients (like the `RemoteCurlClient` and `StorageServiceClient`) can use it with composition / inheritance.

Reply:
why not to use requests.Session?

Reply:
Not really against that in principle, though the added benefits of Session are kinda lost when I'm issuing one-off API requests (and eventually doing so via curl anyway). Still might be worth doing. @bentsi What do you think?

Reply:
Currently, we don't need an implementation of the RestClient using requests lib due to SCT architecture and some legacy code, we need to run the HTTP request directly on the Scylla node. So let's not make this class efficient and invest time in it.